### ISSUE TITLE
allow `MetadataColumn` to convert to `Metadata`

### ISSUE THREAD

**Issue by nbokulich** [qiime2/qiime2#487]
**Improvement Description**
Add a method to `MetadataColumn` to convert to `Metadata`.

**Current Behavior**
There is no direct conversion, so need to use `Metadata(column.to_dataframe())`

**Proposed Behavior**
Either add a method to convert to Metadata, or alternatively MetadataColumn should be substitutable for Metadata in all cases.

**References**
For an example use case, see discussion here: https://github.com/qiime2/q2-sample-classifier/pull/168#discussion_r316852560

**Comment by Oddant1** [comment]
I feel like making `MetadataColumn` suitable as `Metadata` is the preferable solution, but how would this be implemented?

**Comment by ebolyen** [comment]
That's a great question @Oddant1!

For input arguments, we have a transformer step, which let's use do conversions on the fly. However, no such machinery exists for the parameters, as the assumption (up to now) has been that there is only one reasonable representation. I'm not shocked that this no longer holds, I think we always kind of expected to need to convert primitives someday, and this is the most compelling reason so far.

Another thing that is important for this is to retain any sources used when "upcasting" to Metadata so that provenance can be threaded correctly.

---

I think perhaps the simplest approach _at the moment_, is to make a method on MetadataColumn which returns a one-column Metadata object (with originating source tracked, should they exist).

**Comment by thermokarst** [comment]
Does that mean we would/could define transformers for primitives?

**Comment by Oddant1** [comment]
@ebolyen yeah that seems like the optimal band-aid solution until we can make it automatic

**Comment by ebolyen** [comment]
> Does that mean we would/could define transformers for primitives?

Yeah, I think this is probably in our future. :crystal_ball:
